src/main/java/com/bt/code/egress/read/CsvLineMatcher.java

```diff
@@ -0,0 +1,195 @@
+                    sourceColumnIndex, line));
+        } else {
+            LineToken sourceValue = columns[sourceColumnIndex];
+            if (targetColumnIndex >= columns.length) {
+                throw new RuntimeException(String.format("Out of bounds : Could not copy '%s' TO column #%s in row : %s",
+                        sourceValue.getWord(), targetColumnIndex, line));
+            } else {
+                LineToken targetValue = columns[targetColumnIndex];
+                replacement.add(new WordMatch(targetValue, reason, "", sourceValue.getWord()));
+            }
+        }
+        return replacement;
+    }
+
+    private List<WordMatch> createCleanup(LineToken[] columns,
+                                          int targetColumnIndex,
+                                          String line, String reason) {
+        List<WordMatch> cleanup = Lists.newArrayList();
+
+        if (targetColumnIndex >= columns.length) {
+            throw new RuntimeException(String.format("Out of bounds : Could not cleanup column #%s in row : %s",
+                    targetColumnIndex, line));
+        } else {
+            LineToken targetValue = columns[targetColumnIndex];
+            cleanup.add(new WordMatch(targetValue, reason, "", ""));
+            //null for replacement would mean 'no replacement' but empty string means cleanup
+        }
+        return cleanup;
+    }
+
+    private List<WordMatch> createFill(LineToken[] columns,
+                                       int targetColumnIndex,
+                                       String line, String reason, String newValue) {
+        List<WordMatch> fill = Lists.newArrayList();
+
+        if (targetColumnIndex >= columns.length) {
+            throw new RuntimeException(String.format("Out of bounds : Could not fill column #%s in row : %s",
+                    targetColumnIndex, line));
+        } else {
+            LineToken targetValue = columns[targetColumnIndex];
+            fill.add(new WordMatch(targetValue, reason, "", newValue));
+        }
+        return fill;
+    }
+
+    protected List<WordMatch> merge(List<WordMatch> replacements,
+                                  List<WordMatch> cleanups,
+                                  List<WordMatch> fills,
+                                  String line) {
+        Collection<WordMatch> allMatches = union(union(replacements, cleanups), fills);
+        if (CollectionUtils.isEmpty(allMatches)) {
+            return Collections.emptyList();
+        }
+        LinkedList<WordMatch> allMatchesSorted = new LinkedList<>(allMatches);
+        allMatchesSorted.sort(Comparator.comparing(m -> m.getLineToken().getStartPos()));
+        //ha ha who is sure where linked list's last element is ? -)
+        int startPos = allMatchesSorted.getFirst().getLineToken().getStartPos();
+        int endPos = allMatchesSorted.getLast().getLineToken().getEndPos();
```

Comment:
Is it last? sorted by startPos, not endPos

Reply:
CSV replacements config should provide consistent, non-overlapping matches, that produce the same order by startPos and by endPos. I've added code that removes matches that violate that rule